### PR TITLE
Add trailing .0 for double value when needed

### LINKED ISSUE #237: Display DOUBLES  with a decimal point
Doubles with `.0` decimal is currently printed as `integer` it should be displayed as double.
```SQL
CREATE TABLE foo (a DOUBLE NOT NULL, b DOUBLE NOT NULLl);
INSERT INTO foo VALUES {"a": 2.0, "b": 3.1};
SELECT * from foo;
{
  "a": 2,
  "b": 3.1
}
```

expected 

```SQL 
SELECT * from foo;
{
  "a": 2.0,
  "b": 3.1
}
```

### PR DESCRIPTION
**Pull request by yaziine**
This fixes #237.

### CODE REVIEW

File: document/value_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func TestValueString(t *testing.T) {
 		{"bool", document.NewBoolValue(true), "true"},
 		{"int", document.NewIntegerValue(10), "10"},
 		{"double", document.NewDoubleValue(10.1), "10.1"},
+		{"double with trailing .0", document.NewDoubleValue(10), "10.0"},
```
Comment (asdine) [inline document/value_test.go:24]: Could you add a test with a very big float with a trailing .0?

Comment (yaziine) [reply document/value_test.go]: Done, I removed this one and replaced it with a big float that overflows an int64.

Comment (tie) [reply document/value_test.go]: > removed this one

I think we can keep the simple test case.

File: document/value_test.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ func TestValueString(t *testing.T) {
 		{"bool", document.NewBoolValue(true), "true"},
 		{"int", document.NewIntegerValue(10), "10"},
 		{"double", document.NewDoubleValue(10.1), "10.1"},
+		{"double with trailing .0", document.NewDoubleValue(9223372036854775808), "9223372036854775808.0"},
```
Comment (tie) [inline document/value_test.go:24]: Can we also add a test case for `document.NewDoubleValue(1e24)`? Ideally, this should be `1e+24` (note that JSON integers may not have an exponent). See https://play.golang.org/p/AG47LJ-WEE_0

Comment (tie) [reply document/value_test.go:24]: In particular, this would cover `abs ≥ 1e21` branch where `fmt = 'e'`.

Comment (tie) [reply document/value_test.go:24]: Currently FormatFloat returns `1.0e+24` (that’s still a valid JSON number though).
See https://play.golang.org/p/0qpbJPaeoG_G

Comment (tie) [reply document/value_test.go:24]: Actually, `1e21` would be even better since it’s the boundary for exponent format.

Comment (yaziine) [reply document/value_test.go]: Yep good call. I think that it makes sense to have the trailing `.0` even in the `e` notation.

**Review by tie — APPROVED** [review]

**Review by asdine — APPROVED** [review]

**Pull request merged**